crates/types/src/rpc/transactions.rs

```diff
@@ -450,23 +450,28 @@ impl FunctionInvocation {
         // is creating an array with enough room for all objects + 1
         // then based on the order we use this index
 
-        call_info.execution.l2_to_l1_messages.sort_by_key(|msg| msg.order);
-
-        let messages: Vec<OrderedMessageToL1> = call_info
+        let mut messages: Vec<OrderedMessageToL1> = call_info
             .execution
             .l2_to_l1_messages
-            .into_iter()
-            .map(|msg| OrderedMessageToL1::new(msg, call_info.call.caller_address.into()))
+            .iter()
+            .map(|msg| {
+                OrderedMessageToL1::new(
```

Comment:
`OrderedMessageToL1::new` method is used only in this place. Why not refactoring it to accept `&OrderedL2ToL1Message` instead of `OrderedL2ToL1Message`.